front/src/modules/rollingStock/components/RollingStockEditor/PowerRestrictionGridModal.tsx

```diff
@@ -0,0 +1,131 @@
+  useEffect(() => {
+    const modalElement = modalRef.current;
+
+    const focusableElements = modalElement?.querySelectorAll(
+      // last declaration stands for all elements not natively focusable like li
+      'input, button, [tabindex]:not([tabindex="-1"])'
+    ) as NodeListOf<Element>;
+
+    const firstElement = focusableElements[0] as HTMLElement;
+    const lastElement = focusableElements[focusableElements?.length - 1] as HTMLElement;
+
+    /**
+     *
+     * Prevent the tab event and set focus on :
+     * - last element if we are pressing on "shift" in addition to "tab" and are on the first element
+     * - first element if we are only pressing "tab" and are on the last element
+     */
+    const handleTabKeyPress = (event: KeyboardEvent) => {
+      if (event.key === 'Tab') {
+        if (event.shiftKey && document.activeElement === firstElement) {
+          event.preventDefault();
+          lastElement.focus();
+        } else if (!event.shiftKey && document.activeElement === lastElement) {
+          event.preventDefault();
+          firstElement.focus();
+        }
+      }
+    };
+
+    const handleEscapeKeyPress = (event: KeyboardEvent) => {
+      if (event.key === 'Escape') {
+        closeModal();
+      }
+    };
+
+    modalElement?.addEventListener('keydown', handleTabKeyPress);
+    modalElement?.addEventListener('keydown', handleEscapeKeyPress);
+
+    return () => {
+      modalElement?.removeEventListener('keydown', handleTabKeyPress);
+      modalElement?.removeEventListener('keydown', handleEscapeKeyPress);
+    };
+  }, []);
+
+  return (
+    <div className="power-restrictions-modal p-2" ref={modalRef}>
+      <ModalHeaderSNCF withCloseButton withBorderBottom>
+        <InputSNCF
+          id="searchfilter"
+          type="text"
+          value={filter}
+          onChange={(e) => setFilter(e.target.value)}
+          onClear={() => setFilter('')}
+          noMargin
+          unit={<i className="icons-search" />}
+          clearButton
+          focus
+        />
+        {isNewValue && filter !== '' && (
+          <button
+            className="btn btn-sm btn-primary align-self-center ml-3"
+            type="button"
+            onClick={() => {
+              updatePowerRestrictions(filter);
+              closeModal();
+            }}
+          >
+            {t('add')}
+          </button>
+        )}
+      </ModalHeaderSNCF>
+      <ModalBodySNCF>
+        {letterPowerRestrictions && (
+          <Grid
+            data={letterPowerRestrictions}
+            filter={filter}
+            updateData={updatePowerRestrictions}
+            closeModal={closeModal}
+          />
+        )}
+        {digitalPowerRestrictions && (
+          <Grid
+            data={digitalPowerRestrictions}
+            filter={filter}
+            updateData={updatePowerRestrictions}
+            closeModal={closeModal}
```

Comment:
Same here (you can also create a function to do this)
```suggestion
            updateData={(data) => {
              updatePowerRestrictions(data);
              closeModal();
            }}
```
```

Reply:
done (created a function to avoid repeat)